six/modules/c++/six.sicd/include/six/sicd/ComplexClassification.h

```diff
@@ -46,6 +46,16 @@ class ComplexClassification: public Classification
         return os;
     }
 
+    virtual bool equalTo(const Classification& other) const

```

Comment:
When `Classification` is being used, will you always know if you have a `ComplexClassification` or `DerivedClassification`?  I think so... if that's the case, rather than doing this, just use normal `operator==` and take in a `ComplexClassification` here (and don't provide `operator==` in the base class)